### PR TITLE
Update README and general cleanup.

### PR DESCRIPTION
**Pull request by mpaulucci**
Changelog:
- Removed references to zksync-cli
- Removed `make simple_payment` in favor of `cargo run --example simple_payment`
- Updated links that were outdated
- Moved our server url for the CI to github actions.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,8 +9,3 @@ coverage:
 
 test:
 	cargo test -- --test-threads=1
-
-# Examples
-
-simple_payment:
```
Comment (mpaulucci) [inline Makefile:15]: lets use cargo run directly to benefit from clap error messages.

File: README.md [hunk 2]
```diff
@@ -34,16 +34,6 @@ Before you begin, make sure you have the following prerequisites:
 
 - Git: Install Git on your system if you haven't already. You can find installation instructions at [https://git-scm.com/book/en/v2/Getting-Started-Installing-Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).
 
-- zkSync CLI (only if you want to run a localnet): Install the zkSync CLI on your system
```
Comment (mpaulucci) [inline README.md:37]: I added installing local-setup further down, but maybe I should add it here too?

File: envrc.example [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Copy this to `.envrc` to activate.
 
 # Sets the L1 endpoint to be using during test phase.
-export ZKSYNC_WEB3_RS_L1_PROVIDER_URL='http://65.21.140.36:8545'
```
Comment (mpaulucci) [inline envrc.example:4]: we want to use localhost most of the time

File: src/test_utils.rs [hunk 4]
```diff
@@ -13,17 +13,12 @@ pub(crate) struct CompiledContract {
     pub bin: Bytes,
 }
 
-static DEFAULT_L1_PROVIDER_URL: &str = "http://65.21.140.36:8545";
-static DEFAULT_L2_PROVIDER_URL: &str = "http://65.21.140.36:3050";
```
Comment (mpaulucci) [inline src/test_utils.rs:17]: moved this to the CI, otherwise we want to default to localhost

Comment (jpcenteno) [reply src/test_utils.rs:17]: Yes

**Review by jpcenteno — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (ilitteri) [reply README.md:37]: It could be a good idea. It is the same for me.

**Pull request merged**